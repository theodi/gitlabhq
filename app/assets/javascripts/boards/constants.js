export const BoardType = {
  project: 'project',
  group: 'group',
};

export const ListType = {
  assignee: 'assignee',
  milestone: 'milestone',
  backlog: 'backlog',
  closed: 'closed',
  label: 'label',
  promotion: 'promotion',
  blank: 'blank',
};

export const inactiveId = 0;

export const ISSUABLE = 'issuable';
export const LIST = 'list';

export default {
  BoardType,
  ListType,
};
