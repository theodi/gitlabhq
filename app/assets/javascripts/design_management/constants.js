// WARNING: replace this with something
// more sensical as per https://gitlab.com/gitlab-org/gitlab/issues/118611
export const VALID_DESIGN_FILE_MIMETYPE = {
  mimetype: 'image/*',
  regex: /image\/.+/,
};

// https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/types
export const VALID_DATA_TRANSFER_TYPE = 'Files';

export const ACTIVE_DISCUSSION_SOURCE_TYPES = {
  pin: 'pin',
  discussion: 'discussion',
  url: 'url',
};

export const DESIGN_DETAIL_LAYOUT_CLASSLIST = ['design-detail-layout', 'overflow-hidden', 'm-0'];
