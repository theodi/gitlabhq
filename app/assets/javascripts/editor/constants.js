import { __ } from '~/locale';

export const EDITOR_LITE_INSTANCE_ERROR_NO_EL = __(
  '"el" parameter is required for createInstance()',
);

export const URI_PREFIX = 'gitlab';
export const CONTENT_UPDATE_DEBOUNCE = 250;
