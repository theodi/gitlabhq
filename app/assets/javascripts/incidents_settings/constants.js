import { __, s__ } from '~/locale';

/* Integration tabs constants */
export const INTEGRATION_TABS_CONFIG = [
  {
    title: s__('IncidentSettings|Alert integration'),
    component: 'AlertsSettingsForm',
    active: true,
  },
  {
    title: s__('IncidentSettings|PagerDuty integration'),
    component: 'PagerDutySettingsForm',
    active: true,
  },
  {
    title: s__('IncidentSettings|Grafana integration'),
    component: '',
    active: false,
  },
];

export const I18N_INTEGRATION_TABS = {
  headerText: s__('IncidentSettings|Incidents'),
  expandBtnLabel: __('Expand'),
  subHeaderText: s__(
    'IncidentSettings|Set up integrations with external tools to help better manage incidents.',
  ),
};

/* Alerts integration settings constants */

export const I18N_ALERT_SETTINGS_FORM = {
  saveBtnLabel: __('Save changes'),
  introText: __('Action to take when receiving an alert. %{docsLink}'),
  introLinkText: __('More information.'),
  createIssue: {
    label: __('Create an issue. Issues are created for each alert triggered.'),
  },
  issueTemplate: {
    label: __('Issue template (optional)'),
  },
  sendEmail: {
    label: __('Send a separate email notification to Developers.'),
  },
  autoCloseIncidents: {
    label: __('Automatically close incident issues when the associated Prometheus alert resolves.'),
  },
};

export const NO_ISSUE_TEMPLATE_SELECTED = { key: '', name: __('No template selected') };
export const TAKING_INCIDENT_ACTION_DOCS_LINK =
  '/help/operations/metrics/alerts#trigger-actions-from-alerts';
export const ISSUE_TEMPLATES_DOCS_LINK =
  '/help/user/project/description_templates#creating-issue-templates';

/* PagerDuty integration settings constants */

export const I18N_PAGERDUTY_SETTINGS_FORM = {
  introText: s__(
    'PagerDutySettings|Setting up a webhook with PagerDuty will automatically create a GitLab issue for each PagerDuty incident.',
  ),
  activeToggle: {
    label: s__('PagerDutySettings|Active'),
  },
  webhookUrl: {
    label: s__('PagerDutySettings|Webhook URL'),
    helpText: s__(
      'PagerDutySettings|Create a GitLab issue for each PagerDuty incident by %{docsLink}',
    ),
    helpDocsLink: s__('PagerDutySettings|configuring a webhook in PagerDuty'),
    resetWebhookUrl: s__('PagerDutySettings|Reset webhook URL'),
    copyToClipboard: __('Copy'),
    updateErrMsg: s__('PagerDutySettings|Failed to update Webhook URL'),
    updateSuccessMsg: s__('PagerDutySettings|Webhook URL update was successful'),
    restKeyInfo: s__(
      "PagerDutySettings|Resetting the webhook URL for this project will require updating this integration's settings in PagerDuty.",
    ),
  },
  saveBtnLabel: __('Save changes'),
};

export const CONFIGURE_PAGERDUTY_WEBHOOK_DOCS_LINK = 'https://support.pagerduty.com/docs/webhooks';

/* common constants */
export const ERROR_MSG = __('There was an error saving your changes.');
