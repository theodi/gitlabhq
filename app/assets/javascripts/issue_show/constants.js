import { __ } from '~/locale';

export const IssuableStatus = {
  Open: 'opened',
  Closed: 'closed',
};

export const IssuableStatusText = {
  [IssuableStatus.Open]: __('Open'),
  [IssuableStatus.Closed]: __('Closed'),
};

export const IssuableType = {
  Issue: 'issue',
  Epic: 'epic',
  MergeRequest: 'merge_request',
};

export const STATUS_PAGE_PUBLISHED = __('Published on status page');
export const JOIN_ZOOM_MEETING = __('Join Zoom meeting');
