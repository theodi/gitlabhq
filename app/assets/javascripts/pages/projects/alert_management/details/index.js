import AlertDetails from '~/alert_management/details';

document.addEventListener('DOMContentLoaded', () => {
  AlertDetails('#js-alert_details');
});
