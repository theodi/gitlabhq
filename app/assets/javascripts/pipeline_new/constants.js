export const VARIABLE_TYPE = 'env_var';
export const FILE_TYPE = 'file';
