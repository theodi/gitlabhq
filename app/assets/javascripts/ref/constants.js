import { __ } from '~/locale';

export const X_TOTAL_HEADER = 'x-total';

export const SEARCH_DEBOUNCE_MS = 250;

export const DEFAULT_I18N = Object.freeze({
  dropdownHeader: __('Select Git revision'),
  searchPlaceholder: __('Search by Git revision'),
  noResultsWithQuery: __('No matching results for "%{query}"'),
  noResults: __('No matching results'),
  branchesErrorMessage: __('An error occurred while fetching branches.  Retry the search.'),
  tagsErrorMessage: __('An error occurred while fetching tags. Retry the search.'),
  commitsErrorMessage: __('An error occurred while fetching commits.  Retry the search.'),
  branches: __('Branches'),
  tags: __('Tags'),
  commits: __('Commits'),
  noRefSelected: __('No ref selected'),
});
