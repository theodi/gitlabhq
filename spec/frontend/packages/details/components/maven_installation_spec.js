import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { registryUrl as mavenPath } from 'jest/packages/details/mock_data';
import { mavenPackage as packageEntity } from 'jest/packages/mock_data';
import MavenInstallation from '~/packages/details/components/maven_installation.vue';
import CodeInstructions from '~/vue_shared/components/registry/code_instruction.vue';
import { TrackingActions } from '~/packages/details/constants';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('MavenInstallation', () => {
  let wrapper;

  const xmlCodeBlock = 'foo/xml';
  const mavenCommandStr = 'foo/command';
  const mavenSetupXml = 'foo/setup';

  const store = new Vuex.Store({
    state: {
      packageEntity,
      mavenPath,
    },
    getters: {
      mavenInstallationXml: () => xmlCodeBlock,
      mavenInstallationCommand: () => mavenCommandStr,
      mavenSetupXml: () => mavenSetupXml,
    },
  });

  const findCodeInstructions = () => wrapper.findAll(CodeInstructions);

  function createComponent() {
    wrapper = shallowMount(MavenInstallation, {
      localVue,
      store,
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    if (wrapper) wrapper.destroy();
  });

  it('renders all the messages', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('installation commands', () => {
    it('renders the correct xml block', () => {
      expect(
        findCodeInstructions()
          .at(0)
          .props(),
      ).toMatchObject({
        instruction: xmlCodeBlock,
        multiline: true,
        trackingAction: TrackingActions.COPY_MAVEN_XML,
      });
    });

    it('renders the correct maven command', () => {
      expect(
        findCodeInstructions()
          .at(1)
          .props(),
      ).toMatchObject({
        instruction: mavenCommandStr,
        multiline: false,
        trackingAction: TrackingActions.COPY_MAVEN_COMMAND,
      });
    });
  });

  describe('setup commands', () => {
    it('renders the correct xml block', () => {
      expect(
        findCodeInstructions()
          .at(2)
          .props(),
      ).toMatchObject({
        instruction: mavenSetupXml,
        multiline: true,
        trackingAction: TrackingActions.COPY_MAVEN_SETUP,
      });
    });
  });
});
