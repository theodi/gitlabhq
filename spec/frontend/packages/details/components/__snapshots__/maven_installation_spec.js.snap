// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MavenInstallation renders all the messages 1`] = `
<div>
  <h3
    class="gl-font-lg"
  >
    Installation
  </h3>
   
  <p>
    <gl-sprintf-stub
      message="Copy and paste this inside your %{codeStart}pom.xml%{codeEnd} %{codeStart}dependencies%{codeEnd} block."
    />
  </p>
   
  <code-instruction-stub
    copytext="Copy Maven XML"
    instruction="foo/xml"
    label="Maven XML"
    multiline="true"
    trackingaction="copy_maven_xml"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy Maven command"
    instruction="foo/command"
    label="Maven Command"
    trackingaction="copy_maven_command"
    trackinglabel="code_instruction"
  />
   
  <h3
    class="gl-font-lg"
  >
    Registry setup
  </h3>
   
  <p>
    <gl-sprintf-stub
      message="If you haven't already done so, you will need to add the below to your %{codeStart}pom.xml%{codeEnd} file."
    />
  </p>
   
  <code-instruction-stub
    copytext="Copy Maven registry XML"
    instruction="foo/setup"
    label=""
    multiline="true"
    trackingaction="copy_maven_setup_xml"
    trackinglabel="code_instruction"
  />
   
  <gl-sprintf-stub
    message="For more information on the Maven registry, %{linkStart}see the documentation%{linkEnd}."
  />
</div>
`;
